src/components/molecules/EarthEditor/DatasetPane/DatasetModal/Gdrive/index.tsx

```diff
@@ -0,0 +1,145 @@
+
+export type Props = {
+  onReturn: () => void;
+  onSheetSelect: (sheet: SheetParameter) => void;
+  syncLoading: boolean;
+};
+
+const Gdrive: React.FC<Props> = ({ onReturn, syncLoading, onSheetSelect }) => {
+  const intl = useIntl();
+  const theme = useTheme();
+
+  const {
+    isLoading,
+    pickedFile,
+    pickedFileSheets,
+    selectedSheet,
+    handleAuthClick,
+    handleSheetSelect,
+  } = useHooks(onSheetSelect);
+
+  return (
+    <>
+      <StyledIcon icon={"arrowLongLeft"} size={24} onClick={onReturn} />
+      {syncLoading && <Loading />}
+      {pickedFile?.id && (
+        <>
+          <Flex justify="center" align="center">
+            <GdriveIcon size={32} icon="googleDrive" />
+            <Text size="m" color={theme.main.strongText} weight="bold">
+              {intl.formatMessage({ defaultMessage: "Connect with Google Drive" })}
+            </Text>
+          </Flex>
+          <Divider margin="24px" />
+          <Flex justify="center">
+            <AssetCard
+              isImage={false}
+              name={pickedFile.name}
+              cardSize="medium"
+              icon="sheetFile"
+              iconSize="50px"
+              onCheck={() => handleAuthClick()}
+            />
+          </Flex>
+          <Divider margin="24px" />
+          <AssetWrapper direction="column" justify="space-between">
+            <AssetList wrap="nowrap" justify="space-between">
+              {pickedFileSheets?.map((sheetItem: GoolgeSheet) => (
+                <AssetListItem
+                  key={sheetItem.properties.sheetId}
+                  asset={{ id: sheetItem.properties.sheetId, name: sheetItem.properties.title }}
+                  isImage={false}
+                  onCheck={() => {
+                    handleSheetSelect({
+                      id: sheetItem.properties.sheetId,
+                      name: sheetItem.properties.title,
+                    });
+                  }}
+                  selected={sheetItem.properties.sheetId === selectedSheet?.id}
+                />
+              ))}
+            </AssetList>
+          </AssetWrapper>
+          <Divider margin="24px" />
+        </>
+      )}
+
+      {!pickedFile?.id && (
+        <>
+          <Flex justify="center" align="center">
+            <GdriveIcon size={32} icon="googleDrive" />
+            <Text size="m" color={theme.main.strongText} weight="bold">
+              {intl.formatMessage({ defaultMessage: "Connect with Google Drive" })}
+            </Text>
+          </Flex>
+          <Divider margin="24px" />
+          <Flex justify="center">
+            <Text
+              size="m"
+              color={theme.colors.text.weak}
+              otherProperties={{ marginBottom: metricsSizes["m"] + "px" }}>
+              {intl.formatMessage({
+                defaultMessage: "Re:earth supports Google Sheets, and CSV file to upload.",
+              })}
+            </Text>
+          </Flex>
+          <Flex justify="center">
```

Comment:
Is there any way to move this Flex's contents into the Flex just above? Flex is just a div with flexbox so there isn't REALLY any semantic reason to split the contents unless they have different styling needs. 
Actually, instead of the above suggestion can you not remove the Flex above wrapping the single Text component and just do margin left and right auto to the Text component itself? That way we can keep the DOM tree from getting too out of hand.